src/pgduckdb_options.cpp

```diff
@@ -185,7 +154,26 @@ extern "C" {
 DECLARE_PG_FUNCTION(install_extension) {
 	std::string extension_name = pgduckdb::pg::GetArgString(fcinfo, 0);
 	std::string repository = pgduckdb::pg::GetArgString(fcinfo, 1);
-	pgduckdb::DuckdbInstallExtension(extension_name, repository);
+
+	pgduckdb::DuckDBQueryOrThrow(pgduckdb::DuckdbInstallExtensionQuery(extension_name, repository));
+
+	Datum extension_name_datum = CStringGetTextDatum(extension_name.c_str());
+	Datum repository_datum = CStringGetTextDatum(repository.c_str());
+
+	Oid arg_types[] = {TEXTOID, TEXTOID};
+	Datum values[] = {extension_name_datum, repository_datum};
+
+	SPI_connect();
+	auto ret = SPI_execute_with_args(R"(
+		INSERT INTO duckdb.extensions (name, enabled, repository)
+		VALUES ($1, true, $2)
+		ON CONFLICT (name) DO UPDATE SET enabled = true
+		)",
+	                                 lengthof(arg_types), arg_types, values, NULL, false, 0);
+	if (ret != SPI_OK_INSERT)
+		elog(ERROR, "SPI_exec failed: error code %s", SPI_result_code_string(ret));
```

Comment:
```suggestion
	if (ret != SPI_OK_INSERT) {
		elog(ERROR, "SPI_exec failed: error code %s", SPI_result_code_string(ret));
	}
```
also - shouldn't we uninstall the extension? (to maintain a consistent state?